### ISSUE TITLE
RuntimeError: Working outside of application context.

### ISSUE THREAD

**Issue by Tidschimon** [automatic-ripping-machine/automatic-ripping-machine#639]
- [x ] I have checked the wiki
- [x ] I have searched for similar issues

**Describe the bug**
When I started the manual ripping action
```bash
sudo -u arm /usr/bin/python3 /opt/arm/arm/ripper/main.py -d sr0
```
 as part of [General Troubleshooting](https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/General-Troubleshooting), I encountered the Error message:
```
RuntimeError: Working outside of application context.

This typically means that you attempted to use functionality that needed
the current application. To solve this, set up an application context
with app.app_context(). See the documentation for more information.
```
Full stack see the attachments

**To Reproduce**
1. Take a fresh Ubuntu 22.04 install
2. Follow the steps in the [manual setup](https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/Setting-up-ARM-manually-(Ubuntu))
3. On step ["Set up repos and install dependencies"](https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/Setting-up-ARM-manually-(Ubuntu)#set-up-repos-and-install-dependencies) edit the command that adds the repository for ppa:mc3man/focal6 (basically replace 20.04 with 22.04 in the switch case)
4. Follow the rest of the guide
5. The last command will fail with the error message

### Environment
- Ubuntu 22.04
- Arm v2_master

### Log file
[errorstack.txt](https://github.com/automatic-ripping-machine/automatic-ripping-machine/files/10336786/errorstack.txt)

### Investigation
- It seems that a fresh install will install flask at the most recent version (2.2.2 in my case)
- Apparently since v1.1.x flask uses the "application_context" to interact with objects attached to a request or session [flask docu](https://flask.palletsprojects.com/en/1.1.x/appcontext/)
- So basically at more recent versions flask requires one to surround the db call (and most likely everything thereafter) with a 
```python
with app.app_context():
```

I manually added that line and indented everything below and it seems to work now. PR is incoming
Danger Note: I **am** a developer, so I have a general sense in what I am doing, I also know a bit of python but I am neither familiar with the codebase of ARM nor the usage of flask, at all, so be aware...

**Comment by microtechno9000** [comment]
hey @Tidschimon , at the moment Ubuntu 22.04 isn't supported in ARM as there are breaking changes within the various packages.

not sure if @1337-server is working on it at the moment, but it is something we need to update and support as there are multiple issues users have raised with 22.04 recently.

**Comment by Fairchild110** [comment]
Yeah, I'm currently trying to see what I can do to get Alder Lake Quick Sync working with 20.04 and I am just slamming my head against the wall trying to get it to work outside of 22.04.

**Comment by BigPawTate** [comment]
same issue on 20.04

```
discworker@discman:~$ neofetch
            .-/+oossssoo+/-.               discworker@discman
        `:+ssssssssssssssssss+:`           ------------------
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 20.04.5 LTS x86_64
    .ossssssssssssssssssdMMMNysssso.       Host: Inspiron 3647
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 5.4.0-136-generic
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 35 mins
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 990 (dpkg), 4 (snap)
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: bash 5.0.17
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Resolution: 1920x1080
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   Theme: Adwaita [GTK3]
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   Icons: Adwaita [GTK3]
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Terminal: /dev/pts/0
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   CPU: Intel i3-4170 (4) @ 3.700GHz
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    GPU: Intel 4th Generation Core Processor Family
  +sssssssssdmydMMMMMMMMddddyssssssss+     Memory: 328MiB / 3848MiB
   /ssssssssssshdmNNNNmyNMMMMhssssss/
    .ossssssssssssssssssdMMMNysssso.
      -+sssssssssssssssssyyyssss+-
        `:+ssssssssssssssssss+:`
            .-/+oossssoo+/-.

discworker@discman:~$ /usr/bin/python3 /opt/arm/arm/ripper/main.py -d sr0 | at now
warning: commands will be executed using /bin/sh
Traceback (most recent call last):
  File "/opt/arm/arm/ripper/utils.py", line 497, in database_updater
    db.session.commit()
  File "<string>", line 2, in commit
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/orm/scoping.py", line 24, in _proxied
    return self.registry()
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/util/_collections.py", line 1006, in __call__
    key = self.scopefunc()
  File "/usr/local/lib/python3.8/dist-packages/flask_sqlalchemy/session.py", line 81, in _app_ctx_id
    return id(app_ctx._get_current_object())  # type: ignore[attr-defined]
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/local.py", line 513, in _get_current_object
    raise RuntimeError(unbound_message) from None
RuntimeError: Working outside of application context.

This typically means that you attempted to use functionality that needed
the current application. To solve this, set up an application context
with app.app_context(). See the documentation for more information.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/arm/arm/ripper/main.py", line 435, in <module>
    logfile = logger.setup_logging(job)
  File "/opt/arm/arm/ripper/logger.py", line 20, in setup_logging
    logfile = job.logfile = job.identify_audio_cd()
  File "/opt/arm/arm/models/models.py", line 137, in identify_audio_cd
    mb_title = music_brainz.get_title(disc_id, self)
  File "/opt/arm/arm/ripper/music_brainz.py", line 141, in get_title
    u.database_updater(args, job)
  File "/opt/arm/arm/ripper/utils.py", line 504, in database_updater
    raise RuntimeError(str(e))
RuntimeError: Working outside of application context.

This typically means that you attempted to use functionality that needed
the current application. To solve this, set up an application context
with app.app_context(). See the documentation for more information.
job 22 at Sun Jan  8 19:17:00 2023

**Comment by Tidschimon** [comment]
I noticed: Since the ripper's `main.py` differs quite a bit between v2_master and v2_devel, I am not sure if this issue is a problem on the v2_devel, as well. 
Should I open a PR anyways with a big ALERT note that:
- I am not entirely sure if this problem also exists on the v2_devel
- I did not go to the lengths of trying to reproduce it there
- I committed the change in an attempt to re-create a fix that was based on the v2_master branch that worked there

**Comment by shitwolfymakes** [comment]
Closing as stale, update your installation and feel free to reopen if the problem persists!
